components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FirefoxAccountsAuthFeature.kt

```diff
@@ -27,9 +27,21 @@ class FirefoxAccountsAuthFeature(
     private val coroutineContext: CoroutineContext = Dispatchers.Main
 ) {
     fun beginAuthentication() {
+        beginAuthenticationAsync {
+            accountManager.beginAuthenticationAsync().await()
+        }
+    }
+
+    fun beginPairingAuthentication(pairingUrl: String) {
+        beginAuthenticationAsync {
+            accountManager.beginAuthenticationAsync(pairingUrl).await()
+        }
+    }
+
+    private fun beginAuthenticationAsync(beginAuthentication: suspend () -> String) {
         CoroutineScope(coroutineContext).launch {
             val authUrl = try {
-                accountManager.beginAuthenticationAsync().await()
+                beginAuthentication()
             } catch (e: FxaException) {
                 // FIXME return a fallback URL provided by Config...
                 "https://accounts.firefox.com/signin"
```

Comment:
Is this the correct fallback for the pairing flow?

Reply:
Hm this might not be a correct fallback for anything. this will not sign you into the browser.

Reply:
Filed follow-up and updated comment: https://github.com/mozilla-mobile/android-components/issues/2496

Agreed, it's not necessarily the right fallback. We need to be able to differentiate and have different error handling behaviour, for both the default auth and pairing flow.